src/server/conn.rs

```diff
@@ -513,3 +553,165 @@ where
         }
     }
 }
+
+// TODO: different file? idk where to put this.
+mod rewindable {
```

Comment:
@seanmonstar, it looks like hyper as a project prefers to keep new files to a minimum, is this an appropriate place to put this?

Reply:
There isn't an enforced preference, really... keeping things in less files came from two reasons: a) it can get annoying to have to import a bunch of the same stuff over and over, and b) sometimes some types make use of private fields of others (though this is less important now that there exists `pub(scope)`s.

If you think it's clearer in a separate file, that's perfectly fine :)